gems/smithy/lib/smithy/weld.rb

```diff
@@ -62,5 +62,13 @@ def endpoint_function_bindings
     def plugins
       {}
     end
+
+    # Called to construct the protocol plugin. Any protocols defined here will be
+    # merged with other protocols. The key is the protocol trait ID and the value
+    # is the fully qualified class name of the protocol.
+    # @return [Hash<String, String>] protocols for use in protocol plugin.
```

Comment:
A general thought on these - with the `plugins` weld method, it requires us to provide a `require_path` which we then add to the generated client.

Thinking about where protocols come from and how we (for AWS and generic customers) would add them.  This current approach assumes that the class will be available/defined already at runtime.  I think this is reasonable for now, but we may need an approach like plugins where we add a `require`.  

And beyond that we may also want the ability for both plugins and protocols defined by welds to *also* define additional gem dependencies that are added to the gemspec.